pkgs/development/haskell-modules/make-package-set.nix

```diff
@@ -112,7 +112,11 @@ let
       };
     in ps // ps.xorg // ps.gnome2 // { inherit stdenv; } // scopeSpliced;
   defaultScope = mkScope self;
-  callPackage = drv: args: callPackageWithScope defaultScope drv args;
+  callPackage = drv: args: let
+    haskellDrv = callPackageWithScope defaultScope drv args;
+    in haskellDrv // {
+      env = self.shellFor { packages = p: [ haskellDrv ]; };
+    };
```

Comment:
I think this will break if you try to use `(overrideCabal haskellDrv ...).env`. It's probably easiest to either declare this in `passthru` in `generic-builder.nix`, or to use `overrideCabal` here to add it to `passthru`. I'd prefer the former.

Reply:
The problem with the former is that `shellFor` doesn't work with a "raw" derivation, because it depends on `.overrideCabal` being available, which is only added with `callPackage`.

I'll test if this really breaks it.

Reply:
Ah. Then yea adding with `passthru` and `overrideCabal` is important. As is, any use of `overrideCabal` will wipe out this `env`.

Reply:
@Infinisil Actually, I dunno if there's a way to fix this at all. We always want to refer to the final version of the derivation. But `overrideCabal` doesn't give you that. So even if we manage to preserve `env` via `overrideCabal`, it'll be the wrong `env` as soon as anyone does `overrideCabal` again; i.e. if they add a dependency, it won't be added to `env`.

Reply:
So maybe we need to invert the control here: put the source of truth in `generic-builder` and define `shellFor` in terms of that.

Reply:
Actually, how about putting the `extractBuildInputs` thing as a `generic-builder` passthru, define it in terms of that. I think that's the only thing that depends on `overrideCabal`.

Reply:
@Infinisil Yep, sounds good. Eliminating `overrideCabal` entirely from the equation for extracting build inputs sounds like a really good thing.